### ISSUE TITLE
Conf Intervals are computed even if not shown

### ISSUE THREAD

**Issue by zauster** [vincentarelbundock/modelsummary#511]
Hey,

I want to compare a few GLM models (with around ~30 dummy variables), but even the most stripped down version (no t statistics, no VCOV, no GoF), just printing the point estimates takes forever to compute (it actually takes more time to do the model summary than to compute the regressions themselves). From Rprof/summaryRprof it seems that it is the computation of the confidence intervals that takes a lot of time (the functions ci.glm, ci, confint.glm, stats::confint are called one after another). Confidence intervals are however not even part of the output then.

Since I disabled the VCOV option, I assumed that computing the output to be printed would be much faster...

REPREX:

```r
library(data.table)
library(modelsummary)
library(future)
plan(sequential)

N <- 5e4
dprob <- 0.004
dnum <- 10
d_mat <- sample(c(0L, 1L), size = N * dnum, replace = TRUE,
                prob = c(1 - dprob, dprob)) |>
  matrix(ncol = dnum) |>
  as.data.table()
dt_tmp <- data.table(
  depvar = rpois(n = N, lambda = 4),
  value = rnorm(n = N),
  d_mat
)
all_dummies <- paste0("V", 1:dnum)

## 1. very small model
est_formula <- reformulate(
  termlabels = c("value"),
  response = "depvar")
res_1 <- glm(est_formula, family = poisson(), data = dt_tmp)

## 2. model with some dummies
est_formula <- reformulate(
  termlabels = c("value", all_dummies),
  response = "depvar")
res_2 <- glm(est_formula, family = poisson(), data = dt_tmp)

start_sequential <- Sys.time()
tmp_sequential <- tempfile()
Rprof(tmp_sequential)
modelsummary(list(res_1, res_2), output = "markdown",
             statistic = NULL, vcov = NULL, conf_level = NULL,
             gof_omit = "AIC|BIC|RMSE")
Rprof(NULL)
(duration_sequential <- difftime(Sys.time(), start_sequential))

summaryRprof(tmp_sequential)
## finished in ~8 seconds
```

sessionInfo:
```r
> sessionInfo()
R version 4.2.0 (2022-04-22)
Platform: x86_64-pc-linux-gnu (64-bit)
Running under: Arch Linux

Matrix products: default
BLAS/LAPACK: /usr/lib/libopenblas_zenp-r0.3.20.so

locale:
 [1] LC_CTYPE=de_AT.UTF-8       LC_NUMERIC=C              
 [3] LC_TIME=de_AT.UTF-8        LC_COLLATE=de_AT.UTF-8    
 [5] LC_MONETARY=de_AT.UTF-8    LC_MESSAGES=de_AT.UTF-8   
 [7] LC_PAPER=de_AT.UTF-8       LC_NAME=C                 
 [9] LC_ADDRESS=C               LC_TELEPHONE=C            
[11] LC_MEASUREMENT=de_AT.UTF-8 LC_IDENTIFICATION=C       

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
[1] kableExtra_1.3.4         future_1.26.1            modelsummary_0.11.1.9000
[4] here_1.0.1               data.table_1.14.2        conflicted_1.1.0        

loaded via a namespace (and not attached):
 [1] highr_0.9          compiler_4.2.0     parameters_0.18.1  tools_4.2.0       
 [5] digest_0.6.29      viridisLite_0.4.0  memoise_2.0.1      evaluate_0.15     
 [9] lifecycle_1.0.1    checkmate_2.1.0    rlang_1.0.2        cli_3.3.0         
[13] rstudioapi_0.13    parallel_4.2.0     xfun_0.31          fastmap_1.1.0     
[17] stringr_1.4.0      bayestestR_0.12.1  httr_1.4.3         knitr_1.39        
[21] xml2_1.3.3         systemfonts_1.0.4  generics_0.1.2     globals_0.15.1    
[25] webshot_0.5.3      rprojroot_2.0.3    svglite_2.1.0      glue_1.6.2        
[29] performance_0.9.1  listenv_0.8.0      R6_2.5.1           future.apply_1.9.0
[33] parallelly_1.32.0  rmarkdown_2.14     magrittr_2.0.3     scales_1.2.0      
[37] tables_0.9.6       backports_1.4.1    codetools_0.2-18   htmltools_0.5.2   
[41] MASS_7.3-57        datawizard_0.4.1   insight_0.17.1     rvest_1.0.2       
[45] colorspace_2.0-3   stringi_1.7.6      munsell_0.5.0      cachem_1.0.6 
```

**Comment by vincentarelbundock** [comment]
This is a good feature request. I will implement this as soon as possible.

**Comment by vincentarelbundock** [comment]
You know, this might be partially fixed upstream already. Could you try installing the development version of this package:

`remotes::install_github("easystats/parameters")`

Restart R completely, then try again.

**Comment by zauster** [comment]
Thanks, that seems to work fine. E.g. the `ci.glm` function
is now only called when confidence intervals are explicitly
asked for `statistic = 'conf.int'`.

Do you want me to write a unit test, using the reprex from before?

**Comment by zauster** [comment]
The directory `inst/tinytest` is empty in this repo, correct?

**Comment by vincentarelbundock** [comment]
Ah yeah, sorry! I was confusing this with another package of mine. This is just a regular `testthat`

**Comment by zauster** [comment]
How about like this?


```r
library(modelsummary)
library(testthat)

ci_funs <- paste0(c("ci\\.glm", "confint\\.glm", "stats::confint"),
                  collapse = "|")
mod <- list()
mod$Poisson <- glm(am ~ drat, data = mtcars, family = poisson())
mod$Logit <- glm(am ~ qsec, data = mtcars, family = poisson())

test_that(desc = "when confidence intervals are enabled, some CI functions should have been called",
          code = {
            tmp_withCI <- tempfile()
            Rprof(tmp_withCI, interval = 0.005)
            modelsummary(mod, output = "markdown",
                         statistic = "conf.int",
                         gof_omit = "AIC|BIC|RMSE")
            Rprof(NULL)
            rprof_withCI <- summaryRprof(tmp_withCI)
            called_funs <- rownames(rprof_withCI$by.total)
            expect_true(any(grepl(ci_funs, called_funs)))
          })

test_that(desc = "when no confidence intervals are needed, no CI functions should have been called",
          code = {
            tmp_withoutCI <- tempfile()
            Rprof(tmp_withoutCI, interval = 0.005)
            modelsummary(mod, output = "markdown",
                         statistic = NULL, vcov = NULL, conf_level = NULL,
                         gof_omit = "AIC|BIC|RMSE")
            Rprof(NULL)
            rprof_withoutCI <- summaryRprof(tmp_withoutCI)
            called_funs <- rownames(rprof_withoutCI$by.total)
            expect_false(any(grepl(ci_funs, called_funs)))
          })
```

**Comment by vincentarelbundock** [comment]
Thanks, this is great!

I believe the reason `ci.glm` was not called in your 2nd test is because you set `conf_level=NULL`.

I have now added another `grepl` check to see if the `"conf"` string is used in `estimate` or `statistic`. If it isn't, then CIs are not computed. This obviates the need for an explicit `conf_level=NULL`, and should speed things up in a large number of cases.

I added a slightly modified version of your tests to the suite.

https://github.com/vincentarelbundock/modelsummary/commit/2754d18ec19bcd672e0af10bf520fa51eb3cc8b4

Thanks for nudging me in this direction!
